File: AD-HOC/2469/main.js
function mostFrequent() {
    let frequency = {};
    for (let grade of lines[1].split(' ').map(Number)) {
        if (!(grade in frequency)) frequency[grade] = 0;
        frequency[grade]++;
    }
    return Object.entries(frequency).sort((a, b) => b[1] - a[1] || b[0] - a[0])[0][0];
}

let lines = require('fs').readFileSync('/dev/stdin', 'utf-8').split('\n');
console.log(mostFrequent());

File: AD-HOC/2469/main.dart
import 'dart:io';

dynamic mostFrequent() {
  Map<int, int> frequency = {};
  for (int grade in stdin.readLineSync()!.split(' ').map(int.parse)) {
    if (!(frequency.containsKey(grade))) frequency[grade] = 0;
    frequency[grade] = frequency[grade]! + 1;
  }
  return (frequency.entries.toList()
        ..sort((a, b) {
          var diff = b.value.compareTo(a.value);
          if (diff == 0) diff = b.key.compareTo(a.key);
          return diff;
        }))
      .first
      .key;
}

void main() {
  stdin.readLineSync();
  print(mostFrequent());
}
